barman/cloud.py

```diff
@@ -1500,7 +1500,7 @@ def backup(self):
         )
         backup_info.set_attribute("systemid", self.postgres.get_systemid())
         controller = self._create_upload_controller(backup_info.backup_id)
-        if self.postgres.server_version >= 90600 or self.postgres.has_pgespresso:
+        if self.postgres.is_minimal_postgres_version():
```

Comment:
This change makes sense within the context of the PR however the resulting code reads a little odd now. Rather than use `ExclusiveBackupStrategy` if the PostgreSQL version is unsupported we should instead exit with an error. We should probably do this check earlier in the code execution for all Barman Cloud commands.

Reply:
This is a really good point. I will adjust that.